### PR TITLE
discovery table: revalidate race fix

### PR DESCRIPTION
**Pull request by AskAlexSharov**
for https://github.com/ledgerwatch/erigon/issues/11158

### CODE REVIEW

File: p2p/discover/table.go [hunk 1]
```diff
@@ -296,8 +297,9 @@ loop:
 			}
 		case <-revalidateDone:
 			revalidate.Reset(tab.revalidateInterval)
-			if tab.live() == 0 && len(waiting) == 0 && minRefreshTimer == nil {
+			if tab.live() == 0 && len(waiting) == 0 && minRefreshTimerActive.CompareAndSwap(false, true) {
```
Comment (antonis19) [inline p2p/discover/table.go:300]: Does this check also work when `minRefreshTimerActive` is nil ?

Comment (antonis19) [reply p2p/discover/table.go:300]: Should you also remove `minRefreshTimer` now , since it's not doing anything?

Comment (AskAlexSharov) [reply p2p/discover/table.go:300]: it used for `defer timer.stop()`

Comment (antonis19) [reply p2p/discover/table.go:300]: Right, I missed the code below .

**Review by antonis19 — APPROVED** [review]

**Pull request merged**